eng/emitter-package.json

```diff
@@ -1,7 +1,6 @@
 {
   "main": "dist/src/index.js",
   "dependencies": {
-    "@azure-tools/typespec-ts": "0.13.0",
-    "@typespec/versioning": "0.44.0"
+    "@azure-tools/typespec-ts": "latest"
```

Comment:
We have been trying to avoid using tags here as they tend to break things. Is there a plan to ensure publishing a new version doesn't break existing consumers?

Reply:
maybe use ^.

Reply:
@weshaggard Sorry for the late response. Simply speaking, I don't think pin to any version could avoid to break existing consumers considering the emitter uses peerDependencies with range which would always fetch the latest peer dependencies.

- If these breakings are introduced from underlying dependencies for example typespec compiler, and breaks the emitter and finally breaks our end users, then upgrading to the latest emitter would work;
- If these breakings are introduced by emitter for example the supported feature surfaces change, then encuraging users to use the latest would be recommanded.

Reply:
For repeatable builds, we should be pinning the versions of the packages we care about in `package.json`, and also checking in a `package-lock.json` which pins all packages.  I can create PRs for these issues tomorrow, or you can:

https://github.com/Azure/azure-sdk-for-js/issues/26197
https://github.com/Azure/azure-sdk-for-js/issues/26198

Reply:
Thanks for info, pinning all packages would make sense to me. I can take these, just double confirm the `emitter-package-lock.json` would be treated as `package-lock.json` alongside `package.json`, Right?

Reply:
@MaryGao: See my feedback in #26200